### PR TITLE
Introduce a state interface: `posf`, `posg`, `vel` ...

### PR DESCRIPTION
**Pull request by MaximeBouton**
This PR introduces a few functions that define a state interface. 
Instead of using fields: `veh.state.posG`, it uses `posg(veh.state)`

the file `states/interface.jl` has the list of the new functions, they are documented.

Example of how to define a custom state type in the doc.

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -1,6 +1,20 @@
 # About
 
-This is the documentation for AutomotiveDrivingModels.jl. Placeholder for now
+This is the documentation for AutomotiveDrivingModels.jl. 
+
+## Concepts
+
+This section defines a few term that are used accross the package.
```
Comment (mattuntergassmair) [inline docs/src/index.md]: term**s**

File: docs/src/index.md [hunk 2]
```diff
@@ -1,6 +1,20 @@
...
 
-This is the documentation for AutomotiveDrivingModels.jl. Placeholder for now
+This is the documentation for AutomotiveDrivingModels.jl. 
+
+## Concepts
+
+This section defines a few term that are used accross the package. 
+See the specific section of the documentation for a more thorough explanation.
+
+- **Entity**: An entity is a traffic participants that navigates in the environment, it is defined by a physical state (position, velocity, ...), an agent definition (whether it is a car or pedestrian, how large it is, ...), and an ID.
```
Comment (mattuntergassmair) [inline docs/src/index.md]: traffic participant--s--

File: docs/src/states.md [hunk 3]
```diff
@@ -4,15 +4,46 @@ In this section of the documentation we explain the default vehicle state type p
...
-## Agent States
+## Entity state
 
-Agents are represented by  the entity data type provided by `Records.jl`.
+Entities are represented by  the entity data type provided by `Records.jl`.
 The entity data type has three field: a state, a definition, and an id. 
 
 The state of an entity usually describes physical quantity such as position and velocity. 
 
 Two states data structure are provided.
```
Comment (mattuntergassmair) [inline docs/src/states.md]: maybe quickly mention which ones... is it `Vehicle1D` and `VehicleState`?

File: src/collision-checkers/minkowski.jl [hunk 4]
```diff
@@ -381,23 +381,23 @@ function _bounding_radius(veh::Entity{S,D,I}) where {S,D<:AbstractAgentDefinitio
 end
 
 """
-    is_potentially_colliding(A::Entity{VehicleState,D,I}, B::Entity{VehicleState,D,I}) where {D<:AbstractAgentDefinition,I} 
+    is_potentially_colliding(A::Entity{S,D,I}, B::Entity{S,D,I}) where {S,D<:AbstractAgentDefinition,I}
```
Comment (mattuntergassmair) [inline src/collision-checkers/minkowski.jl:384]: Can we assume that every state type `S` has a field called `state`?

File: src/feature-extraction/features.jl [hunk 5]
```diff
@@ -1,26 +1,26 @@
...
+function Base.get(::Feature_VelFs, rec::EntityQueueRecord{S,D,I}, roadway::R, vehicle_index::Int, pastframe::Int=0) where {S,D,I,R}
     veh = rec[pastframe][vehicle_index]
-    FeatureValue(veh.state.v*cos(veh.state.posF.ϕ))
+    FeatureValue(velf(veh.state).s)
 end
-function Base.get(::Feature_VelFt, rec::EntityQueueRecord{S,D,I}, roadway::R, vehicle_index::Int, pastframe::Int=0) where {S<:VehicleState,D,I,R}
+function Base.get(::Feature_VelFt, rec::EntityQueueRecord{S,D,I}, roadway::R, vehicle_index::Int, pastframe::Int=0) where {S,D,I,R}
     veh = rec[pastframe][vehicle_index]
-    FeatureValue(veh.state.v*sin(veh.state.posF.ϕ))
+    FeatureValue(velf(veh.state).t)
```
Comment (mattuntergassmair) [inline src/feature-extraction/features.jl:19]: Niiiiiiiiiice B-)

File: src/states/vehicle_state.jl [hunk 6]
```diff
@@ -23,7 +23,13 @@ VehicleState() = VehicleState(VecSE2(), NULL_FRENET, NaN)
...
 VehicleState(posG::VecSE2{Float64}, roadway::Roadway, v::Float64) = VehicleState(posG, Frenet(posG, roadway), v)
 VehicleState(posG::VecSE2{Float64}, lane::Lane, roadway::Roadway, v::Float64) = VehicleState(posG, Frenet(posG, lane, roadway), v)
-VehicleState(posF::Frenet, roadway::Roadway, v::Float64) = VehicleState(get_posG(posF, roadway), posF, v)
+VehicleState(posF::Frenet, roadway::Roadway, v::Float64) = VehicleState(posg(posF, roadway), posF, v)
+
+posf(veh::VehicleState) = veh.posF
+posg(veh::VehicleState) = veh.posG
+vel(veh::VehicleState) = veh.v
+velf(veh::VehicleState) = (s=veh.v*cos(veh.posF.ϕ), t=s=veh.v*sin(veh.posF.ϕ))
+velg(veh::VehicleState) = (s=veh.v*cos(veh.posG.θ), t=s=veh.v*sin(veh.posG.θ))
```
Comment (mattuntergassmair) [inline src/states/vehicle_state.jl]: I'm confused by the `t=s=veh.v...` syntax - is this a copy paste error?

**Review by mattuntergassmair — CHANGES_REQUESTED** [review]

File: src/collision-checkers/minkowski.jl (hunk 4 above)
Comment (MaximeBouton) [reply src/collision-checkers/minkowski.jl:384]: Every `Entity` has a field called `state`

File: src/states/vehicle_state.jl (hunk 6 above)
Comment (MaximeBouton) [reply src/states/vehicle_state.jl]: copy paste error >< should be fixed now

File: src/collision-checkers/minkowski.jl (hunk 4 above)
Comment (mattuntergassmair) [reply src/collision-checkers/minkowski.jl:384]: Ouh true, my bad, perfect

**Review by raunakbh92 — APPROVED** [review]

**Pull request merged**